lib/ex_aws/kms_client_encryption.ex

```diff
@@ -0,0 +1,52 @@
+defmodule ExAws.KMS.ClientEncryption.Result do
+  @enforce_keys [:encrypted_data_key, :ciphertext]
+  defstruct [:encrypted_data_key, :ciphertext]
+end
+
+defmodule ExAws.KMS.ClientEncryption do
+
+  @doc "Encrypts a map as json with a data key generated for the master key_id provided"
+  @spec encrypt(data :: Map.t(), key_id :: String.t()) :: ExAws.KMS.ClientEncryption.Result
+  def encrypt_map(%{} = data, key_id) do
+    config = ExAws.Config.new(:kms)
+    json = config[:json_codec].encode!(data)
+    encrypt(json, key_id)
+  end
+
+  @doc "Decrypts the result of a call to encrypt_map() (Note: keys will be strings)"
+  @spec decrypt(message :: ExAws.KMS.ClientEncryption.Result) :: Map.t()
+  def decrypt_map(message) do
+    plaintext = decrypt(message)
+    config = ExAws.Config.new(:kms)
+    {:ok, obj} = config[:json_codec].decode(plaintext)
+    obj
+  end
+
+  @doc "Encrypts plaintext with a data key generated for the master key_id provided"
+  @spec encrypt(plaintext :: String.t(), key_id :: String.t()) :: ExAws.KMS.ClientEncryption.Result
+  def encrypt(plaintext, key_id) do
+    case  ExAws.KMS.generate_data_key(key_id, [{:key_spec, "AES_256"}]) |> ExAws.request do
+      {:ok, %{"CiphertextBlob" => encrypted_data_key, "KeyId" => _, "Plaintext" => data_key}} ->
```

Comment:
Also, why is `"AES_256"` hard coded? Is this specified by AWS as the only supported encryption scheme or should we be supporting others?